addons/sale/controllers/portal.py

```diff
@@ -294,7 +294,7 @@ def payment_transaction_token(self, acquirer_id, order_id, save_token=False, acc
         # Create transaction
         vals = {
             'acquirer_id': acquirer_id,
-            'type': order._get_payment_type(),
+            'type': order._get_payment_type('form_save' if save_token else False),
```

Comment:
Nice find :+1: 
It looks like half the payment flows were broken when it comes to tokenization^^